stock_request_purchase/models/stock_rule.py

```diff
@@ -11,14 +11,14 @@ def _prepare_purchase_order_line(self, product_id, product_qty,
                                      product_uom, values, po, supplier):
         vals = super(StockRule, self)._prepare_purchase_order_line(
             product_id, product_qty, product_uom, values, po, supplier)
-        if 'stock_request_id' in values:
-            stock_request = self.env["stock.request"].browse(values['stock_request_id'])
-            vals['stock_request_ids'] = [(4, stock_request.id)]
-            if stock_request.analytic_account_id:
-                vals['account_analytic_id'] = stock_request.analytic_account_id.id
-            if stock_request.analytic_tag_ids:
-                vals['analytic_tag_ids'] = [
-                    (4, tag.id) for tag in stock_request.analytic_tag_ids
+        if "stock_request_id" in values:
+            request = self.env["stock.request"].browse(values["stock_request_id"])
+            vals["stock_request_ids"] = [(4, request.id)]
+            if request._fields.get("analytic_account_id" in request):
```

Comment:
```suggestion
            if request._fields.get("analytic_account_id"):
```
and the same for the other